Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +17,109 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    double totalcurrent= 0;
+    totalcurrent= ( (current1 * ( counter - 1) )/ counter + (current/counter) );
+    return totalcurrent;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
-{
-    return false;
+{   
+    if( current < 0 )
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    double init=    initialStateOfChargePercent_;
+    double avg= ( (current / BATTERY_AMP_HOUR_CAPACITY )) * 100;
+    if(avg <= 0)
+    {
+        avg= -avg;
+    }
+    double averageMin=0; //current/60 will be current per minutes
+    double averageSec=0; //average min /60 will be current per seconds
+    int hours=0;   //variable to multiply by 3600
+    int minutes=0; //variable to multiply by 60
+    int seconds=0; //variable to multiply by 1
+    if(isCharging())
```

Comment:
This is a unique solution, however using these two loops for all these values is pretty intense :p. There's a two line, mathematical solution that I want you to use instead to determine how many milliseconds are left until the battery is charged or depleted.